app/models/enterprise_fee.rb

```diff
@@ -27,7 +27,7 @@ class EnterpriseFee < ActiveRecord::Base
     if user.has_spree_role?('admin')
       scoped
     else
-      where('enterprise_id IN (?)', user.enterprises)
+      where('enterprise_id IN (?)', user.enterprises.select('enterprises.id').map(&:id))
```

Comment:
from #3709 where I found that:
"A scope returns a relation but an association returns a CollectionProxy that has both methods from a relation select and pluck plus all the array methods including map. So we can use map on an association but not on a scope."

enterprises is an association, so you can use pluck here

BUT, why dont you just do select and return a relation, I dont think you need to return an array here.